### PR TITLE
fix: Update service profiles and ports data sources to match corresponding endpoints

### PR DESCRIPTION
**Pull request by thogarty**
* Update schema for service profiles data source
* Update config parsing, request generation, and response handling for service profiles data source
* Fix acceptance tests for service profiles
* Update ports name data source to allow it to return more than one port

### CODE REVIEW

File: equinix/data_source_fabric_service_profile.go [hunk 1]
```diff
@@ -53,66 +54,87 @@ func readFabricServiceProfilesSearchSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Description: "flips view between buyer and seller representation. Available values : aSide, zSide. Default value : aSide",
 		},
+		"and_filters": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Optional boolean flag to indicate if the filters will be AND'd together",
```
Comment (ocobles) [inline equinix/data_source_fabric_service_profile.go:60]: What's the default behavior?

File: docs/data-sources/fabric_service_profiles.md [hunk 2]
```diff
@@ -17,66 +17,81 @@ Additional documentation:
...
+    }
+    sort {
+        direction = "ASC"
+        property = "/name"
+    }
 }
 
-output "id" {
-  value = data.equinix_fabric_service_profile.service_profiles_data_name.data.0.id
+output "number_of_returned_service_profiles" {
```
[the remark this answers is not in the document]
Comment (thogarty) [reply docs/data-sources/fabric_service_profiles.md:42]: The reason they're included is that even though field names are descriptive, the customer base may be so new to Terraform that they'll give feedback looking for these details. We've received that feedback from internal Equinix Terraform customers that would like to have the attribute selection provided to them without needing to learn how to do it from the Terraform Developer docs.

I agree that maintenance is impacted. We can revisit this after the docs generation process has more traction from #685

File: equinix/data_source_fabric_service_profile.go (hunk 1 above)
Comment (thogarty) [reply equinix/data_source_fabric_service_profile.go]: Updated

**Review by srushti-patl — APPROVED** [review]

**Review by jkallem-equinix — DISMISSED** [review]
LGTM

**Review by srushti-patl — APPROVED** [review]

**Pull request merged**